### PR TITLE
Added TransitGatewayRoutes

### PR DESCRIPTION
**Pull request by monty16597**
### APIs
- CreateTransitGatewayRoute
- SearchTransitGatewayRoutes
- DeleteTransitGatewayRoute

### Terraform Test
```terraform
resource "aws_ec2_transit_gateway" "example" {}

resource "aws_ec2_transit_gateway_route_table" "example" {
  transit_gateway_id = aws_ec2_transit_gateway.example.id
}

resource "aws_ec2_transit_gateway_route" "example" {
  destination_cidr_block         = "0.0.0.0/0"
  blackhole                      = true
  transit_gateway_route_table_id = aws_ec2_transit_gateway_route_table.example.id
}

```

## Do not Merge

### CODE REVIEW

File: moto/ec2/models.py [hunk 1]
```diff
@@ -6089,6 +6090,50 @@ def get_all_transit_gateway_route_tables(self, filters):
...
+        transit_gateway_attachment_id=None,
+        blackhole=False
+    ):
+        transit_gateways_route_table = self.transit_gateways_route_tables[transit_gateway_route_table_id]
+        transit_gateways_route_table.routes[destination_cidr_block] = {
+            "destinationCidrBlock": destination_cidr_block,
+            "prefixListId": "random()",
+            "state": "blackhole" if blackhole else "active",
+            "transitGatewayAttachments": {
+                "resourceId": "String",
```
Comment (whummer) [inline moto/ec2/models.py:6106]: Should we add actual values for `resourceId` and `resourceType` here - looks like these values are only placeholders / schema types?

File: moto/ec2/models.py [hunk 2]
```diff
@@ -6089,6 +6090,50 @@ def get_all_transit_gateway_route_tables(self, filters):
...
+    ):
+        transit_gateways_route_table = self.transit_gateways_route_tables[transit_gateway_route_table_id]
+        transit_gateways_route_table.routes[destination_cidr_block]['state'] = "deleted"
+        return transit_gateways_route_table
+
+    def search_transit_gateway_routes(self, transit_gateway_route_table_id, filters, max_results=None):
+        transit_gateway_route_table = self.transit_gateways_route_tables[transit_gateway_route_table_id]
+        routes = []
+        if filters is not None:
+            if filters.get("state") is not None:
```
Comment (whummer) [inline moto/ec2/models.py:6127]: nit: could be simplified to (same in line 6131 below):
```
             if filters.get("state"):
```

File: moto/ec2/models.py [hunk 3]
```diff
@@ -6089,6 +6090,50 @@ def get_all_transit_gateway_route_tables(self, filters):
...
+                for key in transit_gateway_route_table.routes:
+                    if transit_gateway_route_table.routes[key]['state'] in filters.get("state"):
```
Comment (whummer) [inline moto/ec2/models.py:6129]: Can we safely assume that all keys (e.g., `[key]` or `['state']`) are always present in these dicts? Perhaps we should rewrite the code a bit to hedge against `KeyError` (as a general rule, not only related to this PR).

File: moto/ec2/responses/transit_gateway_route_tables.py [hunk 4]
```diff
@@ -29,6 +29,42 @@ def delete_transit_gateway_route_table(self):
         template = self.response_template(DELETE_TRANSIT_GATEWAY_ROUTE_TABLE_RESPONSE)
         return template.render(transit_gateway_route_table=transit_gateway_route_table)
 
+    def create_transit_gateway_route(self):
+        transit_gateway_attachment_id = self._get_param("TransitGatewayAttachmentId")
+        destination_cidr_block = self._get_param("DestinationCidrBlock")
+        transit_gateway_route_table_id = self._get_param("TransitGatewayRouteTableId")
+        blackhole = True if self._get_param("Blackhole") else False
```
Comment (whummer) [inline moto/ec2/responses/transit_gateway_route_tables.py:36]: Could the result of `self._get_param("Blackhole")` also contain the string `"false"`?

Perhaps better change to something like:
```
TRUE_VALUES = [True, "true", "True"]
...
         blackhole = self._get_param("Blackhole") in TRUE_VALUES
```

**Review by whummer — COMMENTED** [review]
Thanks @monty16597 , looks great. 👍 Added a few minor comments.. Thanks

File: moto/ec2/models.py (hunk 1 above)
Comment (monty16597) [reply moto/ec2/models.py:6106]: We need transitGatewayAttachments to achieve this. We will make it dynamic once we implement the TransitGatewayAttachment.

File: moto/ec2/models.py (hunk 2 above)
Comment (monty16597) [reply moto/ec2/models.py:6127]: Sure

File: moto/ec2/responses/transit_gateway_route_tables.py (hunk 4 above)
Comment (monty16597) [reply moto/ec2/responses/transit_gateway_route_tables.py:36]: We can have one util for which will convert str(True or False) to boolean type.

File: moto/ec2/models.py (hunk 1 above)
Comment (whummer) [reply moto/ec2/models.py:6093]: Sounds good - let's please add a `TODO` comment in the code there and indicate in the attribute responses that this is so far still mocked:
```
            "transitGatewayAttachments": {
                # TODO: needs to be fixed once we have support for transit gateway attachments
                "resourceId": "TODO",
                "resourceType": "TODO",
```

**Pull request merged**